File: backend/gcloudTest.py
# from google.cloud import speech
# from google.cloud.speech import enums
# from google.cloud.speech import types
import google.cloud.speech

client = speech.SpeechClient()

# Loads the audio into memory
with open('test.wav', 'rb') as audio_file:
    content = audio_file.read()
    audio = types.RecognitionAudio(content=content)

config = types.RecognitionConfig(
    encoding=enums.RecognitionConfig.AudioEncoding.LINEAR16,
    sample_rate_hertz=16000,
    language_code='en-US')

# Detects speech in the audio file
response = client.recognize(config, audio)

for result in response.results:
    print('Transcript: {}'.format(result.alternatives[0].transcript))

File: backend/gcloudTest.js
async function main() {
  // Imports the Google Cloud client library
  const speech = require('@google-cloud/speech');
  const fs = require('fs');

  // Creates a client
  const client = new speech.SpeechClient();

  // The name of the audio file to transcribe
  const fileName = 'test.wav';

  // Reads a local audio file and converts it to base64
  const file = fs.readFileSync(fileName);
  const audioBytes = file.toString('base64');

  // The audio file's encoding, sample rate in hertz, and BCP-47 language code
  const audio = {
    content: audioBytes,
  };
  const config = {
    encoding: 'LINEAR16',
    sampleRateHertz: 44100,
    languageCode: 'en-US',
    enableWordTimeOffsets: true
  };
  const request = {
    audio: audio,
    config: config,
  };

  // Detects speech in the audio file
  const [response] = await client.recognize(request);
  let words = response.results[0].alternatives[0].words
  for (word of words) {
    startNans = parseInt(word.startTime.seconds) + word.startTime.nanos/1000000000
    endNans = parseInt(word.endTime.seconds) + word.endTime.nanos/1000000000
    console.log(`${word.word}: ${startNans} - ${endNans}`)
  }
}
main().catch(console.error);
